backend/src/auth/auth.controller.ts

```diff
@@ -64,8 +67,8 @@ export class AuthController {
   @ApiOperation({summary: 'Get user profile'})
   @UseGuards(JwtAuthGuard)
   @Get('profile')
-  async getProfile(@Req() req: Request) {
-    return req.user;
+  async getProfile(@GetUser() user: UserDocument) {
```

Comment:
This route isn't working on my end. I keep getting a 401 unauthorized error. I confirmed that I am "logged in" via Swagger authorization. Interestingly, the user route, GET ```/api/users/me```, does work. It might be due to how you're implementing the guard, but I'm not exactly sure why this error is happening. In any case, this route is more of a user route than an auth route, and since both this route and the get user/me route does the same thing, you can choose to remove this one if you wish.